test/t/test_bash.py

```diff
@@ -1,7 +1,19 @@
 import pytest
 
+from conftest import assert_bash_exec
 
 class TestBash:
     @pytest.mark.complete("bash --", require_cmd=True)
     def test_1(self, completion):
         assert completion
+
+    def test_comp_unlocal(self, bash):
+        cmd = ( 'foo() { '
+                      'local VAR=inner; '
+                      '_comp_unlocal VAR; '
+                      'echo $VAR; '
+                '}; '
+                'VAR=outer; foo; unset VAR foo; '
+        )
+        res = assert_bash_exec(bash, cmd, want_output=True).strip()
+        assert res == "outer"
```

Comment:
Thank you for adding the test! The test for the utilities defined in the main `bash_completion` should be placed in `test/t/unit` i.e. `test/t/unit/test_unit_unlocal.py` for this case.